src/sqlfluff/core/rules/std/L036.py

```diff
@@ -79,28 +83,20 @@ def _get_indexes(segment):
             first_new_line_idx,
             first_select_target_idx,
             first_whitespace_idx,
+            select_targets
         )
 
         return eval_result
 
-    @staticmethod
-    def _eval_multiple_select_target_elements(eval_result, segment):
+    def _eval_multiple_select_target_elements(self, eval_result, segment):
         if eval_result.first_new_line_idx == -1:
             # there are multiple select targets but no new lines
-            return LintResult(anchor=segment)
-        else:
```

Comment:
I removed the `else` case because, although this code is indeed executed during the automated tests, removing it seemed to have no ill effects. I think this code may be doing a whitespace check that overlaps with (possibly duplicates?) some of the other whitespace-related rules.

I'd be happy to leave this `else` block in place if there are any concerns about removing it, but it seems cleaner and safer overall to try and avoid having core rules overlap in functionality, because it may make the behavior harder to understand and risks creating mutual undo/redo "loops" between rules if today's overlapping behavior diverges someday.

Reply:
Normally I would say we should keep things like this - but I can't understand what eventuality this piece of code covers so I agree on removing it. There was no discussion in the original PR (#606) about why this was there. @dorzey ?